BigBaseV2/src/hooks/protections/receive_net_message.cpp

```diff
@@ -1,86 +1,60 @@
+		if ((buffer.m_flagBits & 1) == 0 ? (pos = buffer.m_curBit) : (pos = buffer.m_maxBit), length + buffer.m_bitsRead <= pos && buffer.ReadDword((uint32_t*)&msgType, length))
 			return true;
 		else
 			return false;
 	}
-
 	bool hooks::receive_net_message(void* netConnectionManager, void* a2, rage::netConnection::InFrame* frame)
 	{
 		if (frame->get_type() == 4)
 		{
 			rage::datBitBuffer buffer((uint8_t*)frame->m_data, frame->m_length);
 			buffer.m_flagBits = 1;
-			rage::netConnection::MessageType msg_type;
-
+			rage::eNetMessage msgType;
 			const auto player = g_player_service->get_by_msg_id(frame->m_msg_id);
-			if (player && get_message_type(msg_type, buffer))
+			if (player && getMessageType(msgType, buffer))
 			{
-				switch (msg_type)
+				switch (msgType)
 				{
 					//Desync Kick
-					case rage::netConnection::MessageType::MsgNetComplaint:
+					case rage::eNetMessage::CMsgNetComplaint:
 					{
-						uint64_t host_token{};
-						buffer.ReadQWord(&host_token, 64);
-
-						std::vector<CNetGamePlayer*> players;
-
-						uint32_t num_of_host_token{};
-						buffer.ReadDword(&num_of_host_token, 32);
-
-						if (num_of_host_token <= 64) {
-
-							std::vector<uint64_t> host_token_list{};
-							for (uint32_t i = 0; i < num_of_host_token; i++) {
-
-								uint64_t array_element{};
-								buffer.ReadQWord(&array_element, 64);
-								host_token_list.push_back(array_element);
-
-								const auto big_player = g_player_service->get_by_host_token(array_element);
-								if (big_player)
-									if (CNetGamePlayer* net_player = big_player->get_net_game_player())
-										players.push_back(net_player);
-							}
-
-						}
-
+						uint64_t hostToken;
+						buffer.ReadQWord(&hostToken, 0x40);
+						buffer.Seek(0);
+						player_ptr sender = g_player_service->get_by_host_token(hostToken);
+						sender->get_net_game_player()->m_complaints = USHRT_MAX; //Sender
+						g_notification_service->push_warning("Blocked Kick", fmt::format("Desync kick from {}", sender->get_name()));
 						buffer.Seek(0);
-
-						if (!players.empty())
-						{
-							const auto& player = players.at(0);
-							if (player && player->is_valid())
-							{
-								player->m_complaints = 65535;
-								g_notification_service->push_warning("Blocked Kick", std::string("Blocked desync kick from ") + player->get_name());
-							}
-						}
-
 						return false;
 					}
-
+					case rage::eNetMessage::CMsgTextMessage:
+					{
+						char textBuffer[0x100]{};
+						uint64_t peerId{};
+						bool isTeam{};
+						buffer.ReadString(textBuffer, 0x100);
+						buffer.ReadPeerId(&peerId);
+						buffer.ReadBool(&isTeam);
+						g_notification_service->push_warning("Chat", fmt::format("{} [{}]: {}", player->get_name(), isTeam ? HUD::GET_LABEL_TEXT_("MP_CHAT_TEAM") : HUD::GET_LABEL_TEXT_("MP_CHAT_ALL"), textBuffer));
+					} break;
```

Comment:
Remove this code or add options to toggle this, we don't need this to be permanently enabled.